### PR TITLE
enh(UI): Add support for shares in dashboard list

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This adds the support of shares in the dashboard list

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [x] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/front_src/src/Dashboards/Dashboards.cypress.spec.tsx [hunk 1]
```diff
@@ -346,6 +358,47 @@ describe('Dashboards', () => {
...
+      cy.findByTestId('ExpandMoreIcon').click();
+      cy.contains('Kevin').should('be.visible');
+      cy.findByLabelText(labelViewer).should('be.visible');
+      cy.findByLabelText(labelEditor).should('be.visible');
+      cy.findByTestId('PeopleIcon').should('be.visible');
+
+      cy.makeSnapshot();
+    });
+
+    it('revokes the acces right when a row is expanded and ', () => {
```
Comment (bdauria) [inline centreon/www/front_src/src/Dashboards/Dashboards.cypress.spec.tsx:379]:
```suggestion
    it('revokes the access right when a row is expanded and the corresponding action is clicked ', () => {
```

File: centreon/www/front_src/src/Dashboards/api/models.ts [hunk 2]
```diff
@@ -34,6 +34,25 @@ export enum ContactType {
   contactGroup = 'contact_group'
 }
 
+export enum Role {
+  Editor = 'editor',
+  Viewer = 'viewer'
+}
```
Comment (Med1233) [inline centreon/www/front_src/src/Dashboards/api/models.ts:40]: Isn't this the same as the DashboardRole mentioned above ?

File: centreon/www/front_src/src/Dashboards/api/models.ts [hunk 3]
```diff
@@ -34,6 +34,25 @@ export enum ContactType {
...
+export interface Share {
+  contactGroups?: Array<Contact>;
+  contacts?: Array<Contact>;
+}
+
+export enum ShareType {
+  Contact = 'contact',
+  ContactGroup = 'contact_group'
+}
+
```
Comment (Med1233) [inline centreon/www/front_src/src/Dashboards/api/models.ts:55]: Isn't this the same as the ContactType mentioned above

File: centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -16452,4 +16452,13 @@ msgstr "Erweiterter Modus"
...
\ No newline at end of file
+# msgstr ""
+
+# msgid "Delete user"
+# msgstr ""
+
+# msgid "User deleted"
+# msgstr ""
+
+# msgid "You are going to delete <strong>{{name}}</strong> from the user list. He/She won't have access to the dashboard."
```
Comment (smau-centreon) [inline centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po:16463]:
```suggestion
# msgid "You are going to delete <strong>{{name}}</strong> from the user list. This user will no longer access the dashboard."
```

File: centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -16783,4 +16783,13 @@ msgstr "Modo avanzado"
...
\ No newline at end of file
+# msgstr ""
+
+# msgid "Delete user"
+# msgstr ""
+
+# msgid "User deleted"
+# msgstr ""
+
+# msgid "You are going to delete <strong>{{name}}</strong> from the user list. He/She won't have access to the dashboard."
```
Comment (smau-centreon) [inline centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po:16794]:
```suggestion
# msgid "You are going to delete <strong>{{name}}</strong> from the user list. This user will no longer access the dashboard."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -18580,4 +18580,13 @@ msgid "View as cards"
...
\ No newline at end of file
+msgstr "Éditer les propriétés"
+
+msgid "Delete user"
+msgstr "Supprimer un utilisateur"
+
+msgid "User deleted"
+msgstr "Utilisateur supprimé"
+
+msgid "You are going to delete <strong>{{name}}</strong> from the user list. He/She won't have access to the dashboard."
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18591]:
```suggestion
msgid "You are going to delete <strong>{{name}}</strong> from the user list. This user will no longer access the dashboard."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -18580,4 +18580,13 @@ msgid "View as cards"
...
+msgstr "Vous êtes sur le point de supprimer <strong>{{name}}</strong> de la liste des utilisateurs. Il/Elle n'aura plus accès au tableau de bord."
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18592]:
```suggestion
msgstr "Vous êtes sur le point de supprimer <strong>{{name}}</strong> de la liste des utilisateurs. Cet utilisateur n'aura plus accès au tableau de bord."
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -17247,4 +17247,12 @@ msgstr ""
...
-# msgstr ""
\ No newline at end of file
+# msgstr ""
+# msgid "Delete user"
+# msgstr ""
+
+# msgid "User deleted"
+# msgstr ""
+
+# msgid "You are going to delete <strong>{{name}}</strong> from the user list. He/She won't have access to the dashboard."
```
Comment (smau-centreon) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:17257]:
```suggestion
# msgid "You are going to delete <strong>{{name}}</strong> from the user list. This user will no longer access the dashboard."
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -17236,4 +17236,13 @@ msgstr "Você não salvou as alterações, deseja continuar?"
...
\ No newline at end of file
+# msgstr ""
+
+# msgid "Delete user"
+# msgstr ""
+
+# msgid "User deleted"
+# msgstr ""
+
+# msgid "You are going to delete <strong>{{name}}</strong> from the user list. He/She won't have access to the dashboard."
```
Comment (smau-centreon) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:17247]:
```suggestion
# msgid "You are going to delete <strong>{{name}}</strong> from the user list. This user will no longer access the dashboard."
```

File: centreon/www/front_src/src/Dashboards/translatedLabels.tsx [hunk 10]
```diff
@@ -37,3 +37,7 @@ export const labelAccessRightStateUpdated = 'updated';
 export const labelAccessRightsStatsAdded = 'added';
 export const labelAccessRightsStatsRemoved = 'removed';
 export const labelAccessRightsStatsUpdated = 'updated';
+export const labelDeleteUser = 'Delete user';
+export const labelYouAreGoingToDeleteUser =
+  "You are going to delete <strong>{{name}}</strong> from the user list. He/She won't have access to the dashboard.";
```
Comment (smau-centreon) [inline centreon/www/front_src/src/Dashboards/translatedLabels.tsx:42]:
```suggestion
  "You are going to delete <strong>{{name}}</strong> from the user list. This user will no longer access the dashboard.";
```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

**Review by Med1233 — APPROVED** [review]

**Review by Med1233 — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

**Pull request merged**